### PR TITLE
python3Packages.ormar: init at 0.10.23

### PR DESCRIPTION
**Pull request by felix-andreas**
###### Motivation for this change

https://github.com/collerek/ormar

A simple async ORM with fastapi in mind and pydantic validation.

Hi, as this is my first contribution of a Python package, maybe someone else wants to add themselves to the maintainers list?

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/ormar/default.nix [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    sqlalchemy
+    asyncpg
+    psycopg2
+    aiomysql
+    aiosqlite
+    cryptography
+
+    orjson
+    typing-extensions
+    importlib-metadata
```
Comment (fabaff) [inline pkgs/development/python-modules/ormar/default.nix:54]:
```suggestion
  ] ++ lib.optionals (pythonOlder "3.8") [
    typing-extensions
    importlib-metadata
```

see [here](https://github.com/collerek/ormar/blob/master/pyproject.toml#L59-L61) and [here](https://github.com/collerek/ormar/blob/master/pyproject.toml#L63-L65).

File: pkgs/development/python-modules/ormar/default.nix [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "ormar";
+  version = "0.10.23";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "collerek";
+    repo = "ormar";
```
Comment (fabaff) [inline pkgs/development/python-modules/ormar/default.nix:33]: Opinionated change...

```suggestion
    repo = pname;
```

File: pkgs/development/python-modules/ormar/default.nix [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+, codecov
+, pytest-asyncio
+, fastapi
+}:
+
+buildPythonPackage rec {
+  pname = "ormar";
+  version = "0.10.23";
+  format = "pyproject";
+
```
Comment (fabaff) [inline pkgs/development/python-modules/ormar/default.nix:30]: [Python 3.6](https://github.com/collerek/ormar/blob/master/pyproject.toml#L44) is support but I suggest to set the minimal release to 3.7. This saves a condition for `orjson` later and Python 3.6 is EOL technically.

```suggestion
  disabled = pythonOlder "3.7";
```

File: pkgs/development/python-modules/ormar/default.nix [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    aiopg
+    asyncpg
+
+    psycopg2
+    mysqlclient
+    pymysql
+
+    pytestCheckHook
+    pytest-asyncio
+    pytest-cov
```
Comment (fabaff) [inline pkgs/development/python-modules/ormar/default.nix:69]: Usually we try to patch out coverage.

Comment (andreasfelix) [reply pkgs/development/python-modules/ormar/default.nix:71]: Thanks for the fast feedback! What do you exactly mean by patch out? Remove it from the `pyproject.toml`? Do have a link to an exemplary python package where you apply such a patch?

**Review by jonringer — COMMENTED** [review]
Please follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) and [manual#submitting-changes-making-patches ](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) and squash the fix-up commits.

 `git rebase -i` is a powerful command which achieves this, I created a small video demonstrating it's use [here](https://www.youtube.com/watch?v=OR8Q-4PLalc). A more indepth text tutorial can be found [here](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase)

**Comment by andreasfelix** [comment]
Done. I initially intended to wait with squashing the commits until the commit regarding the coverage patch is resolved.

File: pkgs/development/python-modules/ormar/default.nix (hunk 4 above)
Comment (fabaff) [reply pkgs/development/python-modules/ormar/default.nix:71]: No need for patching. `pytest-cov` and `codecov` can simply be removed.

Comment (andreasfelix) [reply pkgs/development/python-modules/ormar/default.nix]: done

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 152630` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.ormar</li>
    <li>python39Packages.ormar</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/152630#issuecomment-1003574808)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-152630/results/python38Packages.ormar/lib/python3.8/site-packages]$ python
Python 3.8.12 (default, Aug 30 2021, 16:42:10) 
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ormar         
>>> ormar.version()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: version() missing 1 required positional argument: 'distribution_name'
```

Looks good to me.

**Pull request merged**